engine/collection/synchronization/engine.go

```diff
@@ -372,72 +334,64 @@ CheckLoop:
 		select {
 		case <-e.unit.Quit():
 			break CheckLoop
-		case <-poll.C:
-			err := e.pollHeight()
-			if err != nil {
-				e.log.Error().Err(err).Msg("could not poll heights")
-			}
-
+		case <-pollChan:
+			e.pollHeight()
 		case <-scan.C:
 			final, err := e.state.Final().Head()
 			if err != nil {
-				e.log.Error().Err(err).Msg("could not get final height")
-				continue
+				e.log.Error().Err(err).Msg("could not get last finalized header")
+				return
 			}
-
-			e.unit.Lock()
 			ranges, batches := e.core.ScanPending(final)
-			err = e.sendRequests(ranges, batches)
-			if err != nil {
-				e.log.Error().Err(err).Msg("could not send requests")
-			}
-			e.unit.Unlock()
+			e.sendRequests(ranges, batches)
 		}
 	}
 
 	// some minor cleanup
 	scan.Stop()
-	poll.Stop()
 }
 
 // pollHeight will send a synchronization request to three random nodes.
-func (e *Engine) pollHeight() error {
-
-	// get the last finalized header
-	final, err := e.state.Final().Head()
+func (e *Engine) pollHeight() {
+	head, err := e.state.Final().Head()
 	if err != nil {
-		return fmt.Errorf("could not get last finalized header: %w", err)
+		e.log.Error().Err(err).Msg("could not get last finalized header")
+		return
 	}
 
 	// send the request for synchronization
 	req := &messages.SyncRequest{
 		Nonce:  rand.Uint64(),
-		Height: final.Height,
+		Height: head.Height,
 	}
-	err = e.conduit.Multicast(req, synccore.DefaultPollNodes, e.participants.NodeIDs()...)
+	err = e.con.Multicast(req, synccore.DefaultPollNodes, e.participants.NodeIDs()...)
 	if err != nil {
-		return fmt.Errorf("could not send sync request: %w", err)
+		e.log.Warn().Err(err).Msg("sending sync request to poll heights failed")
+		return
 	}
 	e.metrics.MessageSent(metrics.EngineClusterSynchronization, metrics.MessageSyncRequest)
-
-	return nil
 }
 
-// sendRequests sends a request for each range and batch.
-func (e *Engine) sendRequests(ranges []flow.Range, batches []flow.Batch) error {
+// sendRequests sends a request for each range and batch using consensus participants from last finalized snapshot.
+func (e *Engine) sendRequests(ranges []flow.Range, batches []flow.Batch) {
+	var errs *multierror.Error
 
-	var errs error
 	for _, ran := range ranges {
 		req := &messages.RangeRequest{
 			Nonce:      rand.Uint64(),
 			FromHeight: ran.From,
 			ToHeight:   ran.To,
 		}
-		err := e.conduit.Multicast(req, synccore.DefaultBlockRequestNodes, e.participants.NodeIDs()...)
+		err := e.con.Multicast(req, synccore.DefaultBlockRequestNodes, e.participants.NodeIDs()...)
 		if err != nil {
-			errs = multierror.Append(errs, fmt.Errorf("could not submit range request (from=%d, to=%d): %w", ran.From, ran.To, err))
```

Comment:
Maybe we could add the more informational error message back (and add to consensus engine as well)